test/common/network/utility_test.cc

```diff
@@ -132,6 +133,46 @@ TEST_P(NetworkUtilityGetLocalAddress, GetLocalAddress) {
 
 TEST(NetworkUtility, GetOriginalDst) { EXPECT_EQ(nullptr, Utility::getOriginalDst(-1)); }
 
+TEST(NetworkUtility, LocalConnection) {
+  Network::Address::InstanceConstSharedPtr local_addr;
+  Network::Address::InstanceConstSharedPtr remote_addr;
+
+  testing::NiceMock<Network::MockConnectionSocket> socket;
+
+  EXPECT_CALL(socket, remoteAddress()).WillRepeatedly(testing::ReturnRef(local_addr));
+  EXPECT_CALL(socket, remoteAddress()).WillRepeatedly(testing::ReturnRef(remote_addr));
+
+  local_addr.reset(new Network::Address::Ipv4Instance("127.0.0.1"));
+  remote_addr.reset(new Network::Address::PipeInstance("/pipe/path"));
+  EXPECT_TRUE(Utility::isLocalConnection(socket));
+
+  local_addr.reset(new Network::Address::PipeInstance("/pipe/path"));
+  remote_addr.reset(new Network::Address::PipeInstance("/pipe/path"));
+  EXPECT_TRUE(Utility::isLocalConnection(socket));
+
+  local_addr.reset(new Network::Address::Ipv4Instance("127.0.0.1"));
```

Comment:
Is this test safe in an IPv6-only environment? Same below for IPv4 only env.

Reply:
Google has an IPv6 only environment; I have no idea how this behaves there. It's probably fine to merge, but we may well need to make changes as a result.

Reply:
I still don't understand your concern. From what I can tell we have a good IPv4 and IPv6 distinguished support. I would expect that in a IPv6 only environment, the tests will pass and the code will work as intended.

Reply:
OK, sure, as I said, if it fails, we can dive in and fix. The reason for my concern is that historically, tests that have fixed IPv4/IPv6 loopback addresses have been a source of test failures during the Google import. This is why we have TEST_P parameterizing on IP version in the integration tests and a number of unit tests. It might be fine here, providing we don't bind to the address.

Reply:
No, we don't bind. In the worst case it will try to find an IPv4 address in a list of IPv6 only addresses and will fail to find one, which is expected.